Frontend/frontend/src/Components/Payments.js

```diff
@@ -54,6 +26,70 @@ export default class Payments extends Component {
         }
     }
 
+    componentDidMount(){
+        // Obtain payments associated with the logged in user.
+        fetch(process.env.REACT_APP_BACKEND_API + "api/Payments/User", {
+            method: "GET",
+            mode: "cors",
+            headers:{
+                Authorization: "Bearer"
+            }
+        }).then(
+            response => {
+                if(response.ok) return response.json()
+            }
+        ).then(
+            APIResult => {
+                // Add user Payments to state.
+                this.setState({
+                    Payments: APIResult
+                });
+            }
+        ).then(() => {
+            // Ensure that all payments are obtained before attempting to get contributors.
+            this.state.Payments.forEach(
+                Payment => {
+                    const RequestURL = `${process.env.REACT_APP_BACKEND_API}api/Payments/Users?paymentId=${Payment["id"]}`;
+                    fetch(RequestURL, {
+                        method: "GET",
+                        mode: "cors",
+                        headers:{
+                            Authorization: "Bearer"
+                        }
+                    }).then(
+                        response => {
+                            if(response.ok) return response.json()
+                        }
+                    ).then(
+                        UsersAPIResult => {
+                            // We need to show only the userName of the contributors.
+                            let user_list = [];
+                            for(let i = 0; i < UsersAPIResult.length; i++) 
+                                user_list.push(UsersAPIResult[i]["userName"]);
+                            this.addUserPayment(user_list);
+                        }
+                    )
+                }
+            )
+        })
+    }
```

Comment:
This should be moved to the APIRequest class in the util folder. 
On line 35 you should probably use the GetAuthString method from the APIRequest class as well. 
If I were to offer a suggestion when this is moved to that APIRequest class I would have a variable called authString which would be put in the header request.
E.g.
`const authString = await APIRequest.getAuthString();`
Then on line 35 and 57, I would change the code to 
`Authorization: authString`

Reply:
Good explanation !

Reply:
API Requests now moved from `Payments.js` to `APIRequests.js` as requested.